### ISSUE TITLE
Error ` Import #0 module="wbg" error: module is not an object or function`

### ISSUE THREAD

**Issue by MichaelMauderer** [rustwasm/wasm-bindgen#2795]
### Summary

After upgrading from wasm-bindgen 0.2.58 to 0.2.78 our WASM application failed to start with the following error.
```
Uncaught (in promise) TypeError: WebAssembly.instantiate(): Import #0 module="wbg" error: module is not an object or function
```

Note that our codebase uses the workaround described in https://github.com/rustwasm/wasm-pack/issues/790, that is we extract the imports generated in `app.js` (glue code) and manually call `WebAssembly.instantiate`/`WebAssembly.instantiateStreaming` on the result. But it does not seem that the error is originating in those calls, according to the stack trace, and the error suggests that this is about a wasm-bindgen related module (the name being `wbg`). 

### The Question
Is this error related to a known issue? Or is there a way to get additional information about the details of what is going on and why the module `wbg` cannot be loaded?

**Comment by wdanilo** [comment]
@alexcrichton sorry for summoning you here, but I know that you are the best source of knowledge and this bug prevents us from developing our product now :( 

I just would love to ask you if you know what could be a potential cause for it. For me, it looks like something changed in the recent was-bindgen release, as the error looks like being thrown by the WASM compiler trying to interpret the results of the module generated by wasm-bindgen. Is that correct?

**Comment by alexcrichton** [comment]
This could unfortunately be coming from a large number of sources. I don't know of anything off the top of my head that would cause this. I'd recommend trying to reduce the bug by removing layers like webpack and/or wasm-pack. If something can be reproduced with just wasm-bindgen I can try to help take a look.

**Comment by wdanilo** [comment]
@alexcrichton thank you so much for the reply! It's already a big help to us if we know that no obvious big breaking changes were made. We will try to reduce it to a minimal example and discover the source of the problem. Thanks again!

**Comment by MichaelMauderer** [comment]
So, we did find the root cause of our problem here. In our patched build system, we were returning the imports generated from the `init` function located in `app.js` and (eventually) passing them to `WebAssembly.instantiate`/`WebAssembly.instantiateStreaming`. At some point, the `init` function was converted from a sync function to an async function, and instead of returning a value, the function returned a promise. This did go well with `WebAssembly.instantiate` and ended in the above-mentioned error.

There is a fun complication to this, that I ended up not further investigating as it now seem to have been red herrings: when trying to minimize the example project I noticed that the error went away if the rust code was reduced to an empty main without imports. But it would appear, again, if `js_sys` was added to the dependencies and imported. Just adding or removing the import `use::js_sys` would toggle the error message.
